packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -215,9 +215,16 @@ class OverlayState extends State<Overlay> {
   }
 
   void _remove(OverlayEntry entry) {
-    setState(() {
-      _entries.remove(entry);
+    assert(() {
+      if (!mounted)
+        assert(_entries.remove(entry));
```

Comment:
That's what Matt initially implemented but I said that was doing pointless work in release mode.